src/main/java/org/opentripplanner/routing/alertpatch/TransitAlert.java

```diff
@@ -109,12 +114,18 @@ public Date getEffectiveStartDate() {
             .orElse(null);
     }
 
+    /**
+     * Finds the last validity endTime from all timePeriods for this alert.
+     * Returns <code>null</code> if the validity is open-ended
+     *
+     * @return Last endDate for this Alert, <code>null</code> if open-ended
+     */
     public Date getEffectiveEndDate() {
         return timePeriods
             .stream()
-            .filter(timePeriod -> timePeriod.endTime > 0)
             .map(timePeriod -> timePeriod.endTime)
             .max(Comparator.naturalOrder())
+            .filter(timePeriod -> timePeriod < TimePeriod.OPEN_ENDED) //If open-ended, null should be returned
```

Comment:
Use `endTime` or just `it` here, not `timePeriod`

Reply:
done